superset/db_engine_specs/base.py

```diff
@@ -142,6 +144,9 @@ class BaseEngineSpec:  # pylint: disable=too-many-public-methods
     ] = None  # used for user messages, overridden in child classes
     _date_trunc_functions: Dict[str, str] = {}
     _time_grain_expressions: Dict[Optional[str], str] = {}
+    _column_type_mappings: Tuple[
+        Tuple[Pattern[str], Union[TypeEngine, Callable[[Match[str]], TypeEngine]]], ...,
+    ] = ()
```

Comment:
Here the type can either be a SQLA type instance, or a callback that receives the match object and in turn returns the SQLA type instance.